mgmt/mcumgr/lib: Remove strnlen usage from OS cmd group and fix how  CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN is used

de-nordic opened the pull request:
The PR contains three commits that:
 1) fix Kconfig information on what `CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN` actually means, as previously it has been stated that it includes string terminating 0, which is not true as CBOR encoding does not include the 0 and the whole length could be used for thread name;
 2) Fixes echo command to not use strnlen making it C99 compliant;
 3) Fixes taskstat command to not use trnlen making it C99 compliant, fixes usage of `CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN` and reduces duplicated code.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

tejlmand commented on subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c:
```diff
@@ -64,13 +65,14 @@ static inline CborError
-	err = cbor_encode_text_string(encoder, thread->name, strnlen(thread->name,
-				      CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN - 1));
-	return err;
+	return cbor_encode_text_stringz(encoder, thread->name);
```
I guess you want to pass `thread_name` here ?

de-nordic replied:
Fixed.

tejlmand commented on subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c:
```diff
@@ -64,13 +65,14 @@ static inline CborError
 				    const struct k_thread *thread)
 {
-	int err = 0;
+	char thread_name[CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN];
+	const int max_size = MIN(ARRAY_SIZE(thread_name), ARRAY_SIZE(thread->name)) - 1;
 
 	ARG_UNUSED(idx);
+	memcpy(thread_name, thread->name, max_size);
+	thread_name[max_size] = '\0';
```
why not just do:
```
size_t name_length = strlen(thread->name);
if (name_length > CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN - 1) {
    name_length = CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN - 1;
}
err = cbor_encode_text_string(encoder, thread->name, name_length);
```

what's the reason for copying to a new array ?

de-nordic replied:
> 
> what's the reason for copying to a new array ?

Hm. Yes, that wasn't the smartest way to do this.

tejlmand reviewed (changes requested):
I believe there is a flaw in code.

And then I have a minor proposal, which won't create an extra array.

de-nordic committed:
mgmt/mcumgr: Kconfig: Correct OS_MGMT_TASKSTAT_THREAD_NAME_LEN info

The help for the OS_MGMT_TASKSTAT_THREAD_NAME_LEN stated that
the config includes terminating zero, but this is not true:
when thread name is encoded with CBOR it is encoded with actual
string length and is not zero terminated.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
mgmg/mcumgr/lib: Remove strnlen from OS echo command

The commit removes usage of strnlen from echo command to make
code C99 compliant, and allow it to compile with newlib.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
mgmt/mcumgr/lib: Fix taskstat thread name processing

The commit:
 - removes usage of strnlen to make taskstat code C99 compliant;
 - corrects thread name len, following update description of
   CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN;
 - reduces duplicated code.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

tejlmand commented on subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c:
```diff
@@ -65,50 +65,40 @@ static inline CborError
 
-	thread_name[sizeof(thread_name) - 1] = 0;
-	ll_to_s((int)thread->base.prio, sizeof(thread_name) - 1, thread_name);
+	if (name_len > (CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN)) {
```
minor nit.

`CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN` is defined as an int, so you may remove the extra set of parenthesis.
https://github.com/zephyrproject-rtos/zephyr/blob/f2dde2819f8fed317ec7cc9432708db7609b0da6/subsys/mgmt/mcumgr/Kconfig#L274-L275

```suggestion
	if (name_len > CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN) {
```

tejlmand reviewed (dismissed):
lgtm, a minor nit observed.

tejlmand reviewed (approved):
lgtm - re-approved
